src/components/directory-item-selector.js

```diff
@@ -28,6 +28,7 @@ const DirectoryItemSelector = (props) => {
             const newNode = {
                 id: newData.elementUuid,
                 name: newData.elementName,
+                type: newData.type,
```

Comment:
do not pass type to TreeViewFinder nodes data which will not be consume in there.
Maybe check what you want by checking if children is undefined or not.

Reply:
Also done